src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionManager.java

```diff
@@ -20,51 +20,130 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import com.google.cloud.bigquery.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import com.hotels.bdp.circustrain.api.CircusTrainException;
+import com.google.cloud.bigquery.Table;
+import com.google.common.annotations.VisibleForTesting;
 
 public class BigQueryDataExtractionManager {
 
-  private final Map<String, BigQueryExtractionData> cache = new HashMap<>();
+  @VisibleForTesting
+  static class TableWrapper {
```

Comment:
I don't understand why we need this wrapper class?

Reply:
Hash Code and Equals in Table object provided by Google is not sufficient, say we load two objects representing the same BigQuery Table and add them to a Set, they will not be considered the same, this was an easy way to implement HashCode + Equals in that manner that I can think of, open to better suggestions